zerver/lib/actions.py

```diff
@@ -3024,6 +3024,15 @@ def do_change_stream_invite_only(stream: Stream, invite_only: bool,
     stream.invite_only = invite_only
     stream.history_public_to_subscribers = history_public_to_subscribers
     stream.save(update_fields=['invite_only', 'history_public_to_subscribers'])
+    event = dict(
+        op="update",
+        type="stream",
+        property="invite_only",
+        value={"invite_only": invite_only,
+               "history_public_to_subscribers": history_public_to_subscribers},
+        stream_id=stream.id,
+    )
+    send_event(event, can_access_stream_user_ids(stream))
```

Comment:
I am not sure that this is right no of users who should receive event. Cause this function returns no of users after stream privacy is changed. Should we remove this and instead add logic to get users on basis of which type of stream it was before. 
If public stream is changed to private, all users should get notified. (We should delete stream, from UI of users who are never subscribed to this stream) 
If private stream is changed to public, all users should get notified as well, but the difference is that non-subscribers won't have stream listed in settings, so we first need to create stream in UI. 
If private stream is changed to other type of private stream, then only subscribers or previously-subscribers should get notified. 

Admins should get notified in all cases. 
Does this make sense?